Refactor variable names in window_guest_stats_bars_paint

aw20368 opened the pull request:
Follow-up to #13146.
- Changed the variable `eax` -> `timeInPark` as suggested
- I found another variable in that function `ebp` that was not descriptive, so I changed it to `barColor`

tupaschoal commented on src/openrct2-ui/windows/Guest.cpp:
```diff
@@ -1317,26 +1317,26 @@ void window_guest_stats_paint(rct_window* w, rct_drawpixelinfo* dpi)
     if (happiness < 10)
         happiness = 10;
-    int32_t ebp = COLOUR_BRIGHT_GREEN;
+    int32_t barColor = COLOUR_BRIGHT_GREEN;
```
Should be `barColour`, as we use British English (applies to whole diff)

aw20368 replied:
As a Canadian, that's a rather embarrassing oversight. Fixed all.

duncanspumpkin commented on src/openrct2-ui/windows/Guest.cpp:
```diff
@@ -1256,7 +1256,7 @@ void window_guest_stats_update(rct_window* w)
  *  rct2: 0x0066ECC1
  *
- *  ebp: colour, contains flag BAR_BLINK for blinking
+ *  colour: contains flag BAR_BLINK for blinking
```
can you look into seeing if this can be refactored into a separate variable. There is no need to have it combined into colour.

aw20368 replied:
Sure, will update.

aw20368 commented:
I've updated the commit with the blink flag refactor.

I also added a commit to refactor the rather ugly (imo) calculations for the guest stats. I left it separate so it can be simply removed from the PR if this isn't desired.

duncanspumpkin commented on src/openrct2-ui/windows/Guest.cpp:
```diff
@@ -1256,10 +1256,9 @@ void window_guest_stats_update(rct_window* w)
  */
 static void window_guest_stats_bars_paint(
-    int32_t value, int32_t x, int32_t y, rct_window* w, rct_drawpixelinfo* dpi, int32_t colour)
+    int32_t value, int32_t x, int32_t y, rct_window* w, rct_drawpixelinfo* dpi, int32_t colour, bool blink_flag)
```
`blinkFlag`

aw20368 replied:
Fixed

duncanspumpkin commented:
All looks about fine.

tupaschoal commented on src/openrct2-ui/windows/Guest.cpp:
```diff
@@ -1314,123 +1320,66 @@ void window_guest_stats_paint(rct_window* w, rct_drawpixelinfo* dpi)
-    window_guest_stats_bars_paint(happiness, screenCoords.x, screenCoords.y, w, dpi, ebp);
+    int32_t happiness = NormalizeGuestStatValue(peep->Happiness, PEEP_MAX_HAPPINESS, 10);
+    int32_t barColour = COLOUR_BRIGHT_GREEN;
+    bool barBlink = (happiness < 50) ? true : false;
```
```suggestion
    bool barBlink = happiness < 50;
```
The `happiness < 50` will already return a bool, no need for the ternary.

tupaschoal commented on src/openrct2-ui/windows/Guest.cpp:
```diff
@@ -1314,123 +1320,66 @@ void window_guest_stats_paint(rct_window* w, rct_drawpixelinfo* dpi)
-    window_guest_stats_bars_paint(energy, screenCoords.x, screenCoords.y, w, dpi, ebp);
+    int32_t energy = NormalizeGuestStatValue(peep->Energy - PEEP_MIN_ENERGY, PEEP_MAX_ENERGY - PEEP_MIN_ENERGY, 10);
+    barColour = COLOUR_BRIGHT_GREEN;
+    barBlink = (energy < 50) ? true : false;
```
```suggestion
    barBlink = energy < 50;
```

tupaschoal commented on src/openrct2-ui/windows/Guest.cpp:
```diff
@@ -1314,123 +1320,66 @@ void window_guest_stats_paint(rct_window* w, rct_drawpixelinfo* dpi)
+    int32_t hunger = NormalizeGuestStatValue(peep->Hunger - 32, 158, 0);
+    hunger = 255 - hunger; // the bar should be longer when peep->Hunger is low
+    barColour = COLOUR_BRIGHT_RED;
+    barBlink = (hunger > 170) ? true : false;
```
```suggestion
    barBlink = hunger > 170;
```

tupaschoal commented on src/openrct2-ui/windows/Guest.cpp:
```diff
@@ -1314,123 +1320,66 @@ void window_guest_stats_paint(rct_window* w, rct_drawpixelinfo* dpi)
+    int32_t thirst = NormalizeGuestStatValue(peep->Thirst - 32, 158, 0);
+    thirst = 255 - thirst; // the bar should be longer when peep->Thirst is low
+    barColour = COLOUR_BRIGHT_RED;
+    barBlink = (thirst > 170) ? true : false;
```
```suggestion
    barBlink = thirst > 170;
```

tupaschoal commented on src/openrct2-ui/windows/Guest.cpp:
```diff
@@ -1314,123 +1320,66 @@ void window_guest_stats_paint(rct_window* w, rct_drawpixelinfo* dpi)
-    window_guest_stats_bars_paint(nausea, screenCoords.x, screenCoords.y, w, dpi, ebp);
+    int32_t nausea = NormalizeGuestStatValue(peep->Nausea - 32, 223, 0);
+    barColour = COLOUR_BRIGHT_RED;
+    barBlink = (nausea > 120) ? true : false;
```
```suggestion
    barBlink = nausea > 120;
```

tupaschoal commented on src/openrct2-ui/windows/Guest.cpp:
```diff
@@ -1314,123 +1320,66 @@ void window_guest_stats_paint(rct_window* w, rct_drawpixelinfo* dpi)
-    window_guest_stats_bars_paint(toilet, screenCoords.x, screenCoords.y, w, dpi, ebp);
+    int32_t toilet = NormalizeGuestStatValue(peep->Toilet - 64, 178, 0);
+    barColour = COLOUR_BRIGHT_RED;
+    barBlink = (toilet > 160) ? true : false;
```
```suggestion
    barBlink = toilet > 160;
```

tupaschoal reviewed (commented):
The `barBlink` logic can be greatly simplified.

I'm not entirely convinced that the `Normalise` method is returning the same thing as before on all the cases, but I'll trust @duncanspumpkin 's judgement

aw20368 commented:
Fixed the `barBlink`. I should've posted my reasoning for the Normalize.

1. Happiness: no scaling, pretty sure this one is correct.
2. Energy:
```
// Before
int32_t energy = ((peep->Energy - PEEP_MIN_ENERGY) * 255) / (PEEP_MAX_ENERGY - PEEP_MIN_ENERGY);
if (energy < 10)
    energy = 10;
```

```
// After
NormalizeGuestStatValue(peep->Energy - PEEP_MIN_ENERGY, PEEP_MAX_ENERGY - PEEP_MIN_ENERGY, 10);

    int32_t newValue = ((peep->Energy - PEEP_MIN_ENERGY) * 255) / (PEEP_MAX_ENERGY - PEEP_MIN_ENERGY);
    return std::clamp(newValue, 10, 255); // same as "before" case since newValue is <= 255
```
3. Hunger, thirst, nausea, and toilet. All four calculations are of the form
```
1    int32_t value = peep->statValue - offset;
2    if (value > max)
3        value = max;
4
5    value -= 32;
6    if (value < 0)
7        value = 0;
8    value *= scalingFactor;
9    value /= 32;
```
The first part (line 1 to 7) is equivalent to `clamp(value - 32, 0, max - 32)`. For lines 8-9, I observed that `scalingFactor` is `floor(255*32/(max - 32))` for all four cases. So lines 8-9 essentially scales the value from `0...(max-32)` to roughly `0...255`, with some rounding error (which I believe/hope is tolerable since this is just calculating the bar length to show up on the UI).

In the Normalize, it corresponds to the above two parts (with the order of the scaling and clamp swapped, the clamp max adjusted accordingly). I then call `Normalize(value - 32, max - 32, 0)` on these cases.

Before (0.3.1):
![image](https://user-images.githubusercontent.com/29646196/96672306-ecacb980-1331-11eb-8ecc-5dbdd3fee9cf.png)

After:
![image](https://user-images.githubusercontent.com/29646196/96672313-f46c5e00-1331-11eb-9a05-492df9bebc99.png)
